doc/guides/build/index.rst

```diff
@@ -0,0 +1,132 @@
+.. _build_overview:
+
+Build Overview
+##############
+
+The Zephyr build process can be divided into two main phases: a
+configuration phase (driven by *CMake*) and a build phase (driven by
+*Make* or *Ninja*). This document is describing the build using *Make*
+as example.
+
```

Comment:
Please add a section about Zephyr modules and link to the corresponding guide. This is an important part of the build process.

Reply:
> Also we have settled on ninja everywhere else when it comes to choosing a build tool. 

This this really matter in this context? Where are not telling people how to use the build system, we describe what happens under the hood.

Reply:
yeah, I'm not seeing how this can improve it. This is documenting how it works under the hood and it is saying that it is using Make as example. Though, if this is really necessary to have it merged I can change it.

Reply:
+1 for Make :-)

Reply:
ninja is the only build tool that works natively on all OSs. Make is the default only on Linux and macOS, not on Windows. All over the documentation we use ninja almost exclusively so that we do not need to handle the Windows case separately.